Formula/git-sync.rb

```diff
@@ -0,0 +1,41 @@
+class GitSync < Formula
+  desc "Clones a git repository and keeps it synchronized with the upstream"
+  homepage "https://github.com/kubernetes/git-sync#readme"
+  url "https://github.com/kubernetes/git-sync/archive/refs/tags/v3.6.0.tar.gz"
+  sha256 "0b65c3f414c98b13d66e99c86680c67348be50d578f88a0888720888b3fcd700"
+  license "Apache-2.0"
+
+  head do
+    url "https://github.com/kubernetes/git-sync.git", branch: "master"
+    depends_on "pandoc" => :build
+  end
+
+  depends_on "go" => :build
+
+  depends_on "coreutils"
+
+  conflicts_with "git-extras", because: "both install `git-sync` binaries"
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    ENV["GOFLAGS"] = "-mod=vendor"
```

Comment:
This isn't needed since go 1.14 (we're coming up on 1.19 now)

Reply:
I set these only because the upstream also sets them as such in their builds. We can't use their build script directly (it is Linux-specific and requires Docker), but the actual build process is trivial, as you can see...

```sh
src/git-sync » gmake
making bin/linux_amd64/git-sync
Unable to find image 'golang:1.18-alpine' locally
1.18-alpine: Pulling from library/golang
530afca65e2e: Pull complete 
d450d4da0343: Pull complete 
8162171ecb65: Pull complete 
9fa0480dbd90: Pull complete 
445ed7c270ad: Pull complete 
Digest: sha256:dda10a0c69473a595ab11ed3f8305bf4d38e0436b80e1462fb22c9d8a1c1e808
Status: Downloaded newer image for golang:1.18-alpine
+ export 'CGO_ENABLED=0'
+ export 'GOARCH=amd64'
+ export 'GOOS=linux'
+ export 'GOFLAGS=-mod=vendor'
+ go list -m
+ go install -installsuffix static -ldflags '-X k8s.io/git-sync/pkg/version.VERSION=v3.6.0' ./...
```

I wanted to deviate from the upstream as little as possible, and these settings do override Go's defaults:
```sh
~ » go env CGO_ENABLED
1
~ » go env GOFLAGS
""
~ » go version
go version go1.18.5 darwin/amd64
 ```

Should I not set them?  It's no problem either way ...

Reply:
Well, the go default is to check for a mod directory called vendor for the past ~4 years. So upstream doesn't need to set this either.

Reply:
Upstream is Kubernetes.. maybe they have their reasons...  It's a no-op for Homebrew for sure... removed in https://github.com/Homebrew/homebrew-core/pull/106907/commits/1335889c04a6419dbd5507ac92fbb1f08f4ad919!